spyder/dependencies.py

```diff
@@ -111,41 +107,31 @@
     {'modname': "jedi",
      'package_name': "jedi",
      'features': _("Main backend for the Python Language Server"),
-     'required_version': JEDI_REQVER}
-]
-
-
-if sys.platform.startswith('linux') and not PY2:
-    DESCRIPTIONS.append(
-        {'modname': "keyring",
-         'package_name': "keyring",
-         'features': _("Save Github credentials to report internal errors securely"),
-         'required_version': KEYRING_REQVER}
-    )
-
-
-DESCRIPTIONS += [
+     'required_version': JEDI_REQVER},
+    {'modname': "keyring",
+     'package_name': "keyring",
+     'features': _("Save Github credentials to report internal "
+                   "errors securely"),
+     'required_version': KEYRING_REQVER,
+     'display': sys.platform.startswith('linux') and not PY2},
     {'modname': "nbconvert",
      'package_name': "nbconvert",
      'features': _("Manipulate Jupyter notebooks in the Editor"),
      'required_version': NBCONVERT_REQVER},
     {'modname': "numpydoc",
      'package_name': "numpydoc",
      'features': _("Improve code completion for objects that use Numpy docstrings"),
-     'required_version': NUMPYDOC_REQVER}
-]
-
-
-if sys.platform == 'nt':
-    DESCRIPTIONS.append(
-        {'modname': "paramiko",
-         'package_name': "paramiko",
-         'features': _("Connect to remote kernels through SSH"),
-         'required_version': PARAMIKO_REQVER}
-    )
-
-
-DESCRIPTIONS += [
+     'required_version': NUMPYDOC_REQVER},
+    {'modname': "paramiko",
+     'package_name': "paramiko",
+     'features': _("Connect to remote kernels through SSH"),
+     'required_version': PARAMIKO_REQVER,
+     'display': sys.platform == 'nt'},

```

Comment:
This needs to be `os.name == 'nt'`. This was an error on my side (I think).

Reply:
Will fix

Reply:
Fixed!